### DISCUSSION TITLE
How to generate different shapes of seeds/grains in Neper

### DISCUSSION THREAD

**Discussion by Ali141272** [neperfepx/neper#572] [Q&A] [answered]
Hello Everyone,
I hope all of you will be fine and doing well.
I have a query that how to generate different shapes of seeds/grains in Neper ( such as in figs a,b,c,d, and e ).
I have generated one model in Neper,  but I need different types of models (just like models in PFC software) and you can find both in the attachment.

Note: I used the below-mentioned commands.
neper -T -n 100 -reg 1 -id 1
neper -T -n 100 -domain "square(0.05,0.1)" -dim 2
neper -M n100-id1.tess -rcl 0.5 -elt tri
neper -V n100-id1.tess -dataedgerad 0.0001 -print img100
Please recommend to me some commands for generating different seeds/grains through Neper software.

Thank You.

<img width="1117" alt="GBS model" src="https://user-images.githubusercontent.com/102291665/193124373-e41699bd-d403-41f3-8018-8ea68ec16d86.png">

**Comment by mkasemer** [comment, marked as answer]
@Ali141272 - There are many resources on the Neper website detailing polycrystal generation with various morphological properties: [here](https://neper.info/generation.html#tessellations-from-morphological-properties) and [here](https://neper.info/doc/tutorials/morpho_statistics.html), for example.

**Reply by Ali141272** [reply]
Hello Respected Sir Matthew Kasemer,
Thank you very much for your kind suggestions. 
I generated different shapes of seeds/grains but when I generate a 3D Model then it shows different shapes of seeds, in addition when I convert it into a 2D model then it gives the same shapes of grains in all three cases. 
The employed command for 2D is (neper -T -n 100 -domain "square(0.05,0.1)" -dim 2)   and for all model generations, I used different commands, which you can see below.

neper -T -n 100 -reg 1
neper -T -n 100 -domain "square(0.05,0.1)" -dim 2
neper -T -n 100 -morpho voronoi
neper -T -n 100 -morpho graingrowth
neper -T -n 100 -morpho "diameq:dirac(1),1-sphericity:lognormal(0.145,0.03)"

Note:  In the case of 3D, when I change the value of long normal (0. 145)  then the shapes of seeds variations take place. 
Please find the enclosed attachment. 

Thank You
<img width="1117" alt="Model" src="https://user-images.githubusercontent.com/102291665/193306377-1c9732d2-0988-4591-abed-aeace689633f.png">

**Reply by mkasemer** [reply]
@Ali141272 : I am having a hard time following the issue. Are you stating that you are attempting to create different morphologies in 2D, but are unable to get anything but a Voronoi tessellation?

**Reply by Ali141272** [reply]
Thank you sir for your kind response,  yes sir I am unable to create different morphologies in 2D.  Suppose I have generated many 3D  models of different morphologies but when I convert them into 2D, then all 3D  models give the same shapes of seeds.

**Reply by rquey** [reply]
@Ali141272 Take the commands you used in 3D and add `-dim 2` to get their 2D equivalents...

**Reply by Ali141272** [reply]
Hello Respected Professor Dr. Romain Quey,
Thank you for your response,  I already used this command (-dim 2). 
If I use this command in the case of 3D then how I can achieve my desired size model in 2D,  because I need a 2D model of size having a width of 0.5m and length of 0.1m.
Professor, please have a look at this command [neper -T -n 100 -domain "square(0.05,0.1)" -dim 2] which I already used for 2D.
